vision/module/location.py

```diff
@@ -346,23 +495,106 @@ def set_img(self, color: np.ndarray, depth: np.ndarray) -> None:
         -------
         None
         """
+        # Set depth and color images
         self.depth = depth
         self.img = color
+
+        # Re-initialize all important values to zero/empty
         self.circles = np.array([])
+        self.clusters = []
+        self.best_cluster = np.array([])
+
+        self.best_slopes = np.array([])
+
         self.center = np.arange(2)
+        self.holes = np.zeros((4, 3), dtype=np.uint16)
+
         self.needs_recalc = True
 
     ## Visualization Functions
 
-    def show_img(self) -> None:
+    def show_img(self, draw_circles: bool = False, draw_center: bool = False) -> None:
         """
-        Shows the initial input image.
+        Displays color image, with color-coded circles and/or center if desired.
+
+        Circle color-coding:
+        -   White & filled in: The best_cluster of circles used for center & in_frame calculation
+        -   Green w/ orange center square: Circles in clusters that have less than 4 circles
+        -   Other colors: Unused clusters that didn't have as many parallel slopes as best_cluster
+
+        Parameters
+        ----------
+        draw_circles: bool
+            Whether to draw detected circles on the image.
+        draw_center: bool
+            Whether to draw the calculated center on the image.
 
         Returns
         -------
         None
         """
-        cv2.imshow("Module Location Image", self.img)
+        circle_img = np.copy(self.img)
+
+        if draw_circles:
+            self.get_center()
```

Comment:
Should this be calling get_center? You would probably want to have the user call get_center and then this function.